File: lec_1/code/row_line.py
matrix = [
    [2, 5, 2],
    [1, 3, 3],
]

vector = [1, 2, 3]


def col_mul(matrix, vector):
    temp_matrix = []

    for i in range(len(vector)):
        vec = [item[i] * vector[i] for item in matrix]
        temp_matrix.append(vec)
    temp = []
    print("Columns by num -> ", temp_matrix)
    for i in range(len(temp_matrix[0])):
        temp.append(sum([item[i] for item in temp_matrix]))
    return temp

# print(col_mul(matrix, vector))


def row_mul(matrix, vector):
    def rows_mul(row_1, row_2): return sum(
        [num_1*num_2 for num_1, num_2 in zip(row_1, row_2)])
    return [rows_mul(item, vector) for item in matrix]


print(row_mul(matrix, vector) == col_mul(matrix, vector))

File: lec_1/code/row_line.js
let matrix = [
  [2, 5],
  [1, 3],
];

let vector = [1, 2];

function col_mul(matrix, vector) {
  let cols = [];

  for (let i = 0; i < matrix.length; i++) {
    let temp = [];
    for (let j = 0; j < matrix[0].length; j++)
      temp.push(matrix[j][i] * vector[i]);
    cols.push(temp);
  }

  matrix = [];
  for (let i = 0; i < cols.length; i++) {
    let row_sum = 0;
    for (let j = 0; j < cols.length; j++) row_sum += cols[j][i];
    matrix.push(row_sum);
  }

  return matrix;
}

function row_mul(matrix, vector) {
  let temp_matrix = [];
  for (let i = 0; i < matrix.length; i++) {
    let num_sum = 0;
    for (let j = 0; j < vector.length; j++) {
      num_sum += matrix[i][j] * vector[j];
    }
    temp_matrix.push(num_sum);
  }
  console.log(temp_matrix);
  return temp_matrix;
}

// col_mul(matrix, vector);
row_mul(matrix, vector);
